src/puppetlabs/puppetdb/jdbc.clj

```diff
@@ -371,28 +371,27 @@
   `(retry-sql* ~n (fn [] ~@body)))
 
 (defn with-transacted-connection-fn
-  "Calls f within a transaction with the specified clojure.jdbc
-  isolation level.  If isolation is nil :read-committed will be used.
+  "Calls f within a transaction with the specified clojure.jdbc isolation level.
+  If isolation is nil, the connection pool default (read committed) is used.
   Retries the transaction up to 5 times."
   [db-spec isolation f]
   (retry-sql 5
              (with-db-connection db-spec
-               (with-db-transaction [:isolation (or isolation :read-committed)]
+               (with-db-transaction [:isolation isolation]
```

Comment:
Offhand,  wonder if it might make make sense to filter out :read-committed too, i.e. if some caller happens to specify it.